test.py

```diff
@@ -1,124 +1,527 @@
+        pl_handler.new_comment(issue_num, payload)
+        pl_handler._find_reviewer.assert_not_called()
+
+
+    def test_new_comment_approved(self):
+        self.github.set_assignee = Mock()
+        self.github.get_labels = Mock(return_value=["S-awaiting-review"])
+        pl_handler = GithubPayloadHandler(self.github)
+        payload = {"issue":{"state":"open","pull_request":1},"comment":{"user":{"login":"bors-servo"},"body":"\br?:@jdm Testing commit"}}
+        issue_num = 1
+
+        pl_handler.new_comment(issue_num, payload)
+        self.github.set_assignee.assert_called_with("jdm", issue_num)
+        self.github.get_labels.assert_called_with(issue_num)
+        self.github.remove_label.assert_called_with("S-awaiting-review", issue_num)
+        self.github.add_label.assert_called_with("S-awaiting-merge")
+
+
+    def test_new_comment_failed_test(self):
+        self.github.set_assignee = Mock()
+        self.github.get_labels = Mock()
+        pl_handler = GithubPayloadHandler(self.github)
+        payload = {"issue":{"state":"open","pull_request":1},"comment":{"user":{"login":"bors-servo"},"body":"\br?:@jdm Test failed"}}
+        issue_num = 1
+
+        pl_handler.new_comment(issue_num, payload)
+        self.github.remove_label.assert_called_with("S-awaiting-merge", issue_num)
+        self.github.add_label.assert_called_with("S-tests-failed", issue_num)
+
+
+    def test_new_comment_merge_conflicts(self):
+        self.github.set_assignee = Mock()
+        self.github.get_labels = Mock()
+        pl_handler = GithubPayloadHandler(self.github)
+        payload = {"issue":{"state":"open","pull_request":1},"comment":{"user":{"login":"bors-servo"},"body":"\br?:@jdm Please resolve the merge conflicts"}}
+        issue_num = 1
+
+        pl_handler.new_comment(issue_num, payload)
+        self.github.remove_label.assert_called_with("S-awaiting-merge", issue_num)
+        self.github.add_label.assert_called_with("S-needs-rebase", issue_num)
+
+
+    @patch('payloadhandler.random')
+    def test_new_pr_no_msg(self, random_mock):
+        self.github.is_new_contributor = Mock(return_value=True)
+        self.github.post_comment = Mock()
+        self.github.get_diff = Mock(return_value="")
+        pl_handler = GithubPayloadHandler(self.github)
+        pl_handler.manage_pr_state = Mock()
+        pl_handler.post_comment = Mock()
+        diff_url = "https://github.com/servo/servo/pull/1234.diff"
+        payload = {"pull_request":{"user":{"login":"jdm"},"diff_url":diff_url}}
+        issue_num = 1
+        rand_val = 'jdm'
+        random_mock.choice.return_value = rand_val
+
+        pl_handler.new_pr(issue_num, payload)
+        pl_handler.manage_pr_state.assert_called_with(issue_num, payload)
+        self.github.post_comment.assert_called_with(GithubPayloadHandler.welcome_msg % rand_val, issue_num)
+        self.github.get_diff.assert_called_with(diff_url)
```

Comment:
Real talk, my eyes glazed over while trying to figure out what this test was actually trying to verify :/ This feels like we're testing at the wrong level to me, when the setup is this involved for what feels like very little payoff.